modin/core/io/text/text_file_dispatcher.py

```diff
@@ -1110,8 +1123,13 @@ def _read(cls, filepath_or_buffer, **kwargs):
             nrows=None,
             compression=compression_infered,
         )
+        fname = partition_kwargs.pop("fname")
```

Comment:
Why do we not want to serialize the func along with partition_kwargs?

Reply:
By doing this separately, we are serializing the function once for the entire workflow, and not every time `read_csv` is called.

Reply:
We could call cls.put(fname) right here rather than calling preprocess_func multiple times in the loop.

Reply:
I just don't like that we call preprocess_func multiple times in the loop.

Reply:
> in the loop.

moved out of the loop